### PR TITLE
DOCS-6663-test-target-variable-cv

### PR DESCRIPTION
**Pull request by Cristian-Venticinque**
Created example for testing processors with target variables

### CODE REVIEW

**Review by egrave — APPROVED** [review]

File: modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -12,7 +12,8 @@
   ** xref:runtime-patching.adoc[Patching the Embedded Mule Runtime Engine Version]
```
Comment (kbostrom) [inline modules/ROOT/nav.adoc:12]: edit instructions
It's a new topic that goes into MUnit testing.
Edit type: CE
Scope: All

File: modules/ROOT/nav.adoc [hunk 2]
```diff
@@ -12,7 +12,8 @@
...
   ** xref:domain-based-application-cookbook.adoc[Test Domain-Based Applications]
   ** xref:test-flow-with-scheduler-cookbook.adoc[Testing a Flow with a Scheduler Source]
-  ** xref:test-recorder-examples.adoc[Recording a Test from an Exchange Example]  
+  ** xref:test-recorder-examples.adoc[Recording a Test from an Exchange Example]
+  ** xref:test-processor-that-uses-target-variable.adoc[Testing a Processor that Uses a Target Variable]
```
Comment (kbostrom) [inline modules/ROOT/nav.adoc]:
```suggestion
  ** xref:test-processor-that-uses-target-variable.adoc[Testing a Processor That Uses a Target Variable]  
```

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 3]
```diff
@@ -0,0 +1,81 @@
+= Testing a Processor that Stores Data in a Target Variable
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc]:
```suggestion
= Testing a Processor That Stores Data in a Target Variable
```
We follow Chicago Manual of style when it comes to title caps: https://capitalizemytitle.com/

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+You can test a processor that stores its result in a target variable by asserting the content of the stored variable.
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc]: Anything to add about why the audience wants or needs to test?

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    doc:name="Request"
+    config-ref="HTTP_Request_configuration"
+    target="httpResponse"
+    targetValue="#[message]"
+    url="http://www.mulesoft.com"/>
+  <logger level="INFO" doc:name="Logger" message="#[vars.httpResponse]"/>
+</flow>
+----
+
+When the HTTP Listener receives a request, the app makes an HTTP Request and the processor stores the received HTTP response in a target variable called `httpResponse`.
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc]: Our term list has the following entry; is it correct?

listener, HTTP listener, HTTP Listener component

Comment (kbostrom) [reply modules/ROOT/pages/test-processor-that-uses-target-variable.adoc]: Also, why capitalize Request? We don't capitalize "response" thereafter....
(and now I see that request is lc at Line 48, so...)

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+<munit:test name="test-target-variable-targetVariableTestFlowTest" description="Test">
+  <munit:execution >
+    <flow-ref doc:name="Flow-ref to targetVariableTestFlow" name="targetVariableTestFlow"/>
+  </munit:execution>
+  <munit:validation >
+    <munit-tools:assert-that doc:name="Assert that" expression="#[vars.httpResponse.attributes.statusCode]" is="#[MunitTools::equalTo(200)]"/>
+  </munit:validation>
+</munit:test>
+----
+
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:47]: The use of "mock" in this narrative was confusing to me, but if it is common in Mule or industry jargon, then STET. I tried to look it up:

https://www.google.com/search?q=mock+when+event+processor&oq=mock+when+event+processor&aqs=chrome..69i57j0i390j69i64l3.6605j0j7&sourceid=chrome&ie=UTF-8

In "normal" English, I think we would use "to create a mock-up" or "create a mock <thing(noun)>."

To "mock" is a **verb** that means primarily "to ridicule" and then "to disappoint the hopes of," "to defy or challenge," and "to imitate closely" (the fourth meaning).

"A mock <thing>" uses "mock" as an **adjective**, to mean a model or sample or imitation.

A "mock-up" is a **noun** for a model for testing: a working sample.

These uses sound better to my ear, but my ear isn't as experienced in Mule-speak as it should be, I suppose!

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+If you want to test the behavior of this app without making the HTTP request, you can mock the processor and the stored target variable.
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:48]: Just an example of what I would expect in "normal" English:
```suggestion
If you want to test the behavior of this app without making the HTTP request, you can create a mock processor and mock stored target variable, and test against them.
```
OR
```suggestion
If you want to test the behavior of this app without making the HTTP request, you can create mock-ups of the processor and the stored target variable and test against them.
```
? Is the test model a model of the app, the request, or the processor and variable?

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+=== Mock a Processor that Stores Data in a Target Variable
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:50]: Just another example:
```suggestion
=== Create a Mock Processor That Stores Data in a Target Variable
```
In this case, we use "mock" as an ADJECTIVE that describes the processor as being an imitation.

Comment (kbostrom) [reply modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:50]: I'll stop commenting about all the different uses of "mock" from here on. You're a better judge of this usage than I am!

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:51]: GLOBALLY standardize capitalization around HTTP Request/request, Listener/listener, Response/response.
Please search; I will stop marking up.

File: modules/ROOT/pages/test-processor-that-uses-target-variable.adoc [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+  </munit:execution>
+  <munit:validation >
+    <munit-tools:assert-that doc:name="Assert that" expression="#[vars.httpResponse.attributes.statusCode]" is="#[MunitTools::equalTo(200)]"/>
+  </munit:validation>
+</munit:test>
+----
+
+== See Also
+
+* xref:mock-event-processor.adoc[Mock When Event Processor]
```
Comment (kbostrom) [inline modules/ROOT/pages/test-processor-that-uses-target-variable.adoc:80]: Does this title mean that there is an event processor named Mock When?

**Review by kbostrom — APPROVED** [review]
CE per request

**Pull request merged**